ipv8/test/test_configuration_pydantic.py

```diff
@@ -0,0 +1,253 @@
+        """
+        self.assertDictEqual(default, dict(IPv8Configuration()))
+
+    def test_format_schema_single(self):
+        """
+        Check if the schema for a single BaseModel is correctly generated.
+        """
+        expected = {
+            "Interface": {
+                "interface": "string",
+                "ip": "string",
+                "port": "integer"
+            }
+        }
+        self.assertDictEqual(expected, format_schema_recursive(Interface))
+
+    def test_format_schema_double(self):
+        """
+        Check if the schema for two BaseModels is correctly generated.
+        """
+        expected = {
+            "Interface": {
+                "interface": "string",
+                "ip": "string",
+                "port": "integer"
+            },
+            "Key": {
+                "alias": "string"
+            }
+        }
+        self.assertDictEqual(expected, format_schema_recursive(Interface, Key))
+
+    def test_format_schema_abstract(self):
+        """
+        Check if the schema for a single BaseModel that inherits from an ABC is correctly generated.
+        """
+        expected = {
+            "FileKey": {
+                "alias": "string",  # Inherited from Key, not defined in FileKey itself
+                "generation": "string",
+                "file": "string"
+            }
+        }
+        self.assertDictEqual(expected, format_schema_recursive(FileKey))
+
+    def test_format_schema_nested(self):
+        """
+        Check if the schema for a single BaseModel that nests another BaseModel is correctly generated.
+        """
+        expected = {
+            "UDPBroadcastBootstrapper": {
+                "class": "string",
+                "init": "#/definitions/BootstrapperInit"
+            },
+            "BootstrapperInit": {}
+        }
+        self.assertDictEqual(expected, format_schema_recursive(UDPBroadcastBootstrapper))
+
+    def test_format_schema_nested_list(self):
+        """
+        Check if the schema for a single BaseModel that nests a list of other BaseModels is correctly generated.
+        """
+        expected = {
+            "BootstrapperInit": {},
+            "Bootstrapper": {"class": "string", "init": "#/definitions/BootstrapperInit"},  # test_format_schema_nested
+            "Walker": {"init": {}, "peers": "integer", "strategy": "string"},
+            "Overlay": {
+                "bootstrappers": ["#/definitions/Bootstrapper"],  # List of nested BaseModels
+                "class": "string",
+                "initialize": {},
+                "key": "string",
+                "on_start": [],
+                "walkers": ["#/definitions/Walker"]  # List of nested BaseModels
+            }
+        }
+        self.assertDictEqual(expected, format_schema_recursive(Overlay))
+
+    def test_format_schema_enum(self):
+        """
+        Check if schemas for Enums (also allowed inside of pydantic BaseModels) are correctly generated.
```

Comment:
"of" looks a bit redundant here
```suggestion
        Check if schemas for Enums (also allowed inside pydantic BaseModels) are correctly generated.
```